.circleci/config.yml

```diff
@@ -56,21 +66,73 @@ jobs:
             - target
           key: cargocache-v2-storage-plus:1.64.0-{{ checksum "~/project/Cargo.lock" }}
 
-  lint:
+  build_minimal:
+    docker:
+      - image: rustlang/rust:nightly
+    working_directory: ~/project/
+    steps:
+      - checkout:
+          path: ~/project
+      - run:
+          name: Version information
+          command: rustc --version; cargo --version; rustup --version; rustup target list --installed
+      - run:
+          name: Remove Cargo.lock
+          command: rm Cargo.lock
+      - restore_cache:
+          keys:
+            - cargocache-v2-storage-plus:1.64.0-minimal-{{ checksum "~/project/Cargo.toml" }}
+      - run:
+          name: Build library for native target (with iterator and macro)
+          command: cargo build -Zminimal-versions --all-features
+      - run:
+          name: Run unit tests (with iterator and macro)
+          command: cargo test --workspace -Zminimal-versions --all-features
+      - save_cache:
+          paths:
+            - /usr/local/cargo/registry
+            - target
+          key: cargocache-v2-storage-plus:1.64.0-minimal-{{ checksum "~/project/Cargo.toml" }}
+
+  build_maximal:
     docker:
       - image: rust:1.64.0
+    working_directory: ~/project/
     steps:
-      - checkout
+      - checkout:
+          path: ~/project
       - run:
           name: Version information
           command: rustc --version; cargo --version; rustup --version; rustup target list --installed
       - run:
-          name: Generate a lockfile
+          name: Update Cargo.lock
           command: cargo update
+      - restore_cache:
+          keys:
+            - cargocache-v2-storage-plus:1.64.0-{{ checksum "~/project/Cargo.lock" }}
+      - run:
+          name: Build library for native target (with iterator and macro)
+          command: cargo build --locked --all-features
+      - run:
+          name: Run unit tests (with iterator and macro)
+          command: cargo test --workspace --locked --all-features
+      - save_cache:
+          paths:
+            - /usr/local/cargo/registry
+            - target
+          key: cargocache-v2-storage-plus:1.64.0-{{ checksum "~/project/Cargo.lock" }}
```

Comment:
We just use `{{ checksum "Cargo.lock" }}` in cosmwasm

Reply:
Sounds good